From f9c43f7bc8edff2e9accc993adb87cd0b5719980 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Wed, 31 Aug 2016 09:38:12 +0100
Subject: [PATCH 1/5] Replace deprecated readdir_r with readdir
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

readdir_r is deprecated and should be replaced with readdir. readdir_r
is considered unsafe on systems where NAME_MAX is undefined and on
some systems it cannot read directory entries with very long names.

Fixes build warning:
topology.c:357:2: warning: ‘readdir_r’ is deprecated
  [-Wdeprecated-declarations]
  while (!readdir_r(dir, &dirent, &direntp)) {
  ^~~~~
In file included from topology.c:35:0:
/usr/include/dirent.h:183:12: note: declared here
 extern int readdir_r (DIR *__restrict __dirp,

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 topology.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/topology.c b/topology.c
index 39b07bd..e293a59 100644
--- a/topology.c
+++ b/topology.c
@@ -331,7 +331,7 @@ static struct cpu_topology *topo_folder_scan(char *path, folder_filter_t filter)
 {
 	DIR *dir, *dir_topology;
 	char *basedir, *newpath;
-	struct dirent dirent, *direntp;
+	struct dirent *direntp;
 	struct stat s;
 	int ret;
 	int is_online;
@@ -354,11 +354,7 @@ static struct cpu_topology *topo_folder_scan(char *path, folder_filter_t filter)
 		return result;
 	}
 
-	while (!readdir_r(dir, &dirent, &direntp)) {
-
-		if (!direntp)
-			break;
-
+	while ((direntp = readdir(dir)) != NULL) {
 		if (direntp->d_name[0] == '.')
 			continue;
 
-- 
2.9.3

